pdns/dnsdist.cc

```diff
@@ -1158,9 +1165,9 @@ static bool prepareOutgoingResponse(LocalHolders& holders, ClientState& cs, DNSQ
   return true;
 }
 
-ProcessQueryResult processQuery(DNSQuestion& dq, ClientState& cs, LocalHolders& holders, std::shared_ptr<DownstreamState>& selectedBackend)
-{
+bool sendReplyCheck(DNSQuestion& dq, ClientState& cs, LocalHolders& holders, std::shared_ptr<DownstreamState>& selectedBackend, time_t* cacheValidTime){
```

Comment:
Split check destination function into 2 separated function:
1.sendReplyCheck - check if we can reply to user query
2.sendtoBackend - check if we can send this request to backend
This is because the original function didn't support both actions simultaneously.